From 784754340a972b44e9835c74e8466f865392785d Mon Sep 17 00:00:00 2001
From: Tomasz Figa <t.figa@samsung.com>
Date: Wed, 24 Apr 2013 19:02:54 +0200
Subject: [PATCH 0328/1302] clk: samsung: exynos4: Do not disable ISP bus
 clocks

Some ISP bus clocks must be kept enabled when entering sleep mode,
otherwise the system hangs. This patch adds CLK_IGNORE_UNUSED flags to
those clocks to keep them ungated when FIMC IS driver is not present.

Signed-off-by: Tomasz Figa <t.figa@samsung.com>
Signed-off-by: MyungJoo Ham <myungjoo.ham@samsung.com>
---
 drivers/clk/samsung/clk-exynos4.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/clk/samsung/clk-exynos4.c b/drivers/clk/samsung/clk-exynos4.c
index 9994b8d..adb1352 100644
--- a/drivers/clk/samsung/clk-exynos4.c
+++ b/drivers/clk/samsung/clk-exynos4.c
@@ -847,13 +847,13 @@ struct samsung_gate_clock exynos4x12_gate_clks[] __initdata = {
 			E4X12_GATE_IP_PERIR, 16, 0, 0, "keypad"),
 	GATE(tmu_apbif, "tmu_apbif", "aclk100", E4X12_GATE_IP_PERIR, 17, 0, 0),
 	GATE(sclk_pwm_isp, "sclk_pwm_isp", "div_pwm_isp",
-			E4X12_SRC_MASK_ISP, 0, CLK_SET_RATE_PARENT, 0),
+			E4X12_SRC_MASK_ISP, 0, CLK_SET_RATE_PARENT | CLK_IGNORE_UNUSED, 0),
 	GATE(sclk_spi0_isp, "sclk_spi0_isp", "div_spi0_isp_pre",
-			E4X12_SRC_MASK_ISP, 4, CLK_SET_RATE_PARENT, 0),
+			E4X12_SRC_MASK_ISP, 4, CLK_SET_RATE_PARENT | CLK_IGNORE_UNUSED, 0),
 	GATE(sclk_spi1_isp, "sclk_spi1_isp", "div_spi1_isp_pre",
-			E4X12_SRC_MASK_ISP, 8, CLK_SET_RATE_PARENT, 0),
+			E4X12_SRC_MASK_ISP, 8, CLK_SET_RATE_PARENT | CLK_IGNORE_UNUSED, 0),
 	GATE(sclk_uart_isp, "sclk_uart_isp", "div_uart_isp",
-			E4X12_SRC_MASK_ISP, 12, CLK_SET_RATE_PARENT, 0),
+			E4X12_SRC_MASK_ISP, 12, CLK_SET_RATE_PARENT | CLK_IGNORE_UNUSED, 0),
 	GATE(pwm_isp_sclk, "pwm_isp_sclk", "sclk_pwm_isp",
 			E4X12_GATE_IP_ISP, 0, 0, 0),
 	GATE(spi0_isp_sclk, "spi0_isp_sclk", "sclk_spi0_isp",
-- 
1.8.3.2

